counsel.el

```diff
@@ -5394,6 +5432,46 @@ specified by the `blddir' property."
             :action #'counsel-compile--action
             :caller 'counsel-compile))
 
+
+;(defun ivy--format-function-generic (selected-fn other-fn cands separator)
+(defun counsel-compile-env--format-hint (cands)
+  "Return a formatter for compile-env CANDS."
+  (ivy--format-function-generic
+   (lambda (selected)
+     (format "%s %s"
+             (if (member selected counsel-compile-env)
+                 (propertize "remove" 'face 'font-lock-warning-face)
+               (propertize "add" 'face
+                           'font-lock-variable-name-face))
```

Comment:
I wonder if these calls to `propertize`, which cons new strings each time they're called, should be hoisted as lexical vars outside of the `lambda` - or is that not worth it?